core/src/eval/mod.rs

```diff
@@ -956,12 +956,17 @@ impl<R: ImportResolver, C: Cache> VirtualMachine<R, C> {
                     slf.reset();
                 }
                 Ok(t) => match t.as_ref() {
-                    Term::Array(ts, _) => {
+                    Term::Array(ts, attrs) => {
                         for t in ts.iter() {
                             // After eval_closure, all the array elements  are
                             // closurized already, so we don't need to do any tracking
                             // of the env.
-                            inner(slf, acc, t.clone());
+                            let value_with_ctr = RuntimeContract::apply_all(
```

Comment:
Oh, so we did evaluate arrays, but just didn't apply lazy contracts?

Reply:
Yep, so `[ "a" | Number ]` was working before, but not `["a"] | Array Number`